src/encoded/static/components/viz/Workflow/parsing-functions.js

```diff
@@ -46,10 +177,10 @@ export function parseAnalysisSteps(analysis_steps, parsingMethod = 'output'){
     }
 
     function compareTwoFilesByUUID(file1, file2){
```

Comment:
We currently still allow `steps.inputs.run_data.file` to be in form of either a UUID or an embedded object.
We could simplify to only handle embedded object to make this front-end code _slightly_ simpler, after transforming 'file' to a linkTo and ensuring its consistent embeddedness, however there might be a benefit to this type-ambiguousness if publishing to NPM, as others' data/references might differ.